File: Graph/Graph_1260.cpp
#include <iostream>
#include <cstring>
#include <vector>
#include <queue>
#include <algorithm>

using namespace std;

vector<int> edge[1001];
bool check[1001];

void dfs(int start)
{
    check[start] = true;
    printf("%d ", start);

    for(int i=0; i < edge[start].size(); i++)
    {
        int next = edge[start][i];
        if(check[next] == false)
        {
            dfs(next);
        }
    }
}

void bfs(int start)
{
    memset(check, false, sizeof(check));
    queue<int> q;

    check[start] = true;
    q.push(start);

    while(!q.empty())
    {
        int node = q.front();
        q.pop();
        printf("%d ", node);

        for(int i=0; i < edge[node].size(); i++)
        {
            int next = edge[node][i];
            if(check[next] == false)
            {
                check[next] = true;
                q.push(next);
            }
        }
    }
}

int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    int v, e, start;
    cin >> v >> e >> start;

    for(int i=0; i < e; i++)
    {
        int n, m;
        cin >> n >> m;

        edge[n].push_back(m);
        edge[m].push_back(n);
    }

    for(int i=0; i <= v; i++)
    {
        sort(edge[i].begin(), edge[i].end());
    }

    dfs(start);
    puts("");
    bfs(start);
    puts("");
}

File: Graph/Graph_1260.py
# Baekjoon 1260

def dfs(x):
    check[x] = True;
    print(x, end=' ')

    for i in range(0, len(edge[x])):
        if check[edge[x][i]] != True:
            dfs(edge[x][i])

def bfs(x):
    queue = []
    queue.append(x)
    check[x] = True

    while len(queue) != 0:
        parents = queue.pop(0)
        print(parents, end=' ')

        for i in range(0, len(edge[parents])):
            if check[edge[parents][i]] == False:
                check[edge[parents][i]] = True
                queue.append(edge[parents][i])
    

v, e, start = map(int, input().split())

check = [False for _ in range(v+1)]

edge = [[] for _ in range(v+1)]

for __ in range(e):
    n1, n2 = map(int, input().split())
    edge[n1].append(n2)
    edge[n2].append(n1)

for i in range(len(edge)):
    edge[i].sort()

dfs(start)

check = [False for _ in range(v+1)]

print()

bfs(start)